### PR TITLE
Predicted Backups

### LINKED ISSUE #335: Show backup boxes and predicted schedule on backups page when backup in slot does not exist

### PR DESCRIPTION
**Pull request by abemassry**
screen shot 2016-08-31 at 7 57 45 am

screen shot 2016-08-31 at 7 55 41 am

Closes #335

### CODE REVIEW

**Comment by na3d** [comment]
Make sure that if you have 5+ backups that it doesn't pop outside to the right.

**Comment by eatonphil** [comment]
> Make sure that if you have 5+ backups that it doesn't pop outside to the right.

This is not history, you have 4 slots.

File: src/linodes/layouts/BackupsPage.js [hunk 1]
```diff
@@ -249,16 +247,146 @@ export class BackupsPage extends Component {
...
-        <p>
-          No backups yet. First automated backup is scheduled for {next.fromNow(true)} from now.
-        </p>
-      );
+      futureBackups[0] = {};
+      futureBackups[0].created = 'Snapshot';
+      futureBackups[0].content = "You haven't taken any snapshots yet";
+      futureBackups[0].id = 'this should never be selected 0';
+    }
+    const timeslot = (timeslot) => {
```
Comment (SirCmpwn) [inline src/linodes/layouts/BackupsPage.js]: Would be neater if this were a dict.

File: src/linodes/layouts/BackupsPage.js [hunk 2]
```diff
@@ -249,16 +247,146 @@ export class BackupsPage extends Component {
...
+
+    const getBackupDayOfWeek = (backupType) => {
+      if (schedule.dayOfWeek) {
+        if (schedule.dayOfWeek === 'Scheduling') {
+          backupDay = 'TBD';
+        } else {
+          if (backupType === 'Daily') {
+            backupDay = moment()._locale
+                        ._weekdays[getNextBackupDay(schedule.timeOfDay)
+                                                   .weekday()];
```
Comment (SirCmpwn) [inline src/linodes/layouts/BackupsPage.js]: Weird indentation here, can you clean this up?

**Comment by SirCmpwn** [comment]
:+1:

File: src/linodes/layouts/BackupsPage.js (hunk 1 above)
Comment (eatonphil) [inline src/linodes/layouts/BackupsPage.js]: 👍

**Comment by eatonphil** [comment]
👍  after @SirCmpwn's comments

**Comment by na3d** [comment]
👍  LGTM

**Comment by abemassry** [comment]
Thanks! 🎉 💯 🏁

**Pull request merged**